realis.go

```diff
@@ -1059,3 +1060,50 @@ func (r *realisClient) RollbackJobUpdate(key aurora.JobUpdateKey, message string
 	}
 	return resp, nil
 }
+
+// VariableBatchStep returns the current batch the update is in during a variable batch update.
+func (r *realisClient) VariableBatchStep(key aurora.JobUpdateKey) (int, error) {
+
+	// Query Aurora for an Update that is either paused or rolling forward
+	resp, err := r.JobUpdateDetails(aurora.JobUpdateQuery{
+		UpdateStatuses: []aurora.JobUpdateStatus{
+			aurora.JobUpdateStatus_ROLLING_FORWARD,
+			aurora.JobUpdateStatus_ROLL_FORWARD_PAUSED,
+		},
+		Key:   &key,
+		Limit: 1,
+	})
+
+	if err != nil {
+		return 0, errors.Wrap(err, "unable to determine current batch the update is in")
+	}
+
+	jobUpdates := response.JobUpdateDetails(resp)
+	if len(jobUpdates) == 0 {
+		return 0, errors.New("update provided could not be found in Aurora")
+	}
+
+	strategy := jobUpdates[0].GetUpdate().GetInstructions().GetSettings().GetUpdateStrategy()
+	if !strategy.IsSetVarBatchStrategy() {
+		return 0, errors.New("update strategy used by update config is not supported by this function")
+	}
+
+	// Using a map to create a set to guard against multiple INSTANCE_UPDATING events for the same shard.
```

Comment:
would we have multiple INSTANCE_UPDATING for a shard?

Reply:
Not according to my tests, but guarding against any future changes and/or unwelcome surprises.